### ISSUE TITLE
Find a mentor to add for an ECT at your school using TRN and NiNo

### ISSUE THREAD

**Issue by mason-emily** [DFE-Digital/register-early-career-teachers#764]
## Description
This story will provide the functionality for school users to find the mentor they are trying to add using a national insurance number (combined with TRN) when date of birth doesn't match a TRS record. 

## User story

**As** school user
**I need** to find a mentor's record in TRS
**So that** I can continue the journey of registering the right mentor and assigning them to an ECT at my school

## Prototype, screenshots & content
[Prototype mentor journey
](https://ecf2-prototy-v4-k7mv50fp1jl0ue.herokuapp.com/v4/school/mentor/what-youll-need)

[Prototype asking for National Insurance number](https://ecf2-prototy-v4-k7mv50fp1jl0ue.herokuapp.com/v4/school/mentor/nino)

**Screenshots of relevant screens:**
Note, the 'corrected name' question is in the prototype but not in scope for this ticket - will be added later.
![image](https://github.com/user-attachments/assets/25e1602c-8dc9-413b-bdef-41d8412c0293)

Content should be as per the prototype.
URL and validation messages should be as per [this doc](https://educationgovuk.sharepoint.com/:x:/r/sites/TeacherServices/Shared%20Documents/Teacher%20Continuing%20Professional%20Development/Teacher%20CPD%20Team/Register%20early%20career%20teachers/Beta/Design/Register%20ECTs%20page%20titles%20and%20URLs.xlsx?d=w0c1578ea643147b29357c7f610ee178f&csf=1&web=1&e=U8iIdi).

## Acceptance criteria

- [x] 1. When user submits a valid TRN and date of birth on the <ins>'Find a mentor'</ins> page, if the check for a TRS record on TRN and date of birth combined fails to find a mentor's record, user is navigated to the <ins>'We cannot find the mentor's details'</ins> page.
- [x] 2. On the <ins>'We cannot find the mentor's details'</ins> page:
    - [x] 2a. User is asked to give a National insurance number for that mentor.
    - [x] 2b. Validation and error massages for the NINO field should be as per [this doc](https://educationgovuk.sharepoint.com/:x:/r/sites/TeacherServices/Shared%20Documents/Teacher%20Continuing%20Professional%20Development/Teacher%20CPD%20Team/Register%20early%20career%20teachers/Beta/Design/Register%20ECTs%20page%20titles%20and%20URLs.xlsx?d=w0c1578ea643147b29357c7f610ee178f&csf=1&web=1&e=U8iIdi).
    - [x] 2c. There will be a 'Back' link which takes the user back to the previous page.
- [x] 3. When user has entered a valid NINO and clicks 'Continue', the NINO is used with TRN to try and find the mentor's TRS record.
- [x] 4. If the mentor's TRS record is found, user progresses to the <ins>'Check mentor details'</ins> page (see #759).
    - [x] 4a. On the 'Check mentor details' page, National Insurance number will be shown in the table ~~as well as~~ instead of the date of birth.     
    - [x] 4b. The <ins>'Check mentor details'</ins> page will otherwise work in the same way as the existing page for when there is an initial match on TRN and date of birth.
- [ ] 5. If the mentor's TRS record can't be found, user is navigated to the <ins>'We were unable to match the mentor with the details you provided'</ins> page.
    - [ ] 5a. On the <ins>'We were unable to match the mentor with the details you provided'</ins> page, the text 'reviewing their teacher record' will be a hyperlink to https://check-a-teachers-record.education.gov.uk/check-records/sign-in and the text 'Find a lost TRN service' will be a hyperlink to https://find-a-lost-trn.education.gov.uk/start as per the prototype. Both links should navigate the user within the same tab.
    - [x] 5b. There will be a green 'Try again' button which navigates the user to the <ins>'Find a mentor'</ins> page where the fields should be reset to blank.
    - [x] 5c. There will be no 'Back' link on the <ins>'We were unable to match the mentor with the details you provided'</ins> page.

## Technical notes
A very similar flow has been built for the ECT journey.

## Linked tickets
Dependent on #759 

Equivalent in the ECT journey #512

**Comment by gracecbrennan** [comment]
Just started and in progress - likelt to be ready EOD

**Comment by gracecbrennan** [comment]
@TobyRet to review

**Comment by ltello** [comment]
[Review app](https://cpd-ec2-review-840-web.test.teacherservices.cloud/)

**Comment by gracecbrennan** [comment]
@JSJohal did you manage to take a look at this one? 

@TobyRet has approved from a dev perspective

**Comment by JSJohal** [comment]
Reviewed and checked off most of the acceptance criteria. Looks good!
One thing to update: content / paragraphs / full stops etc on this page isn't aligned with the [prototype (p: ecf2)](https://ecf2-prototype-e3c7aba62d82.herokuapp.com/v5/other-pages/add-mentor/not-found)
<img width="883" alt="image" src="https://github.com/user-attachments/assets/915d2fb2-2ecc-45c4-88a0-a1b67cf24fc2">

**Comment by ltello** [comment]
Fixed now.
